core/metrics.go

```diff
@@ -52,7 +51,9 @@ var (
 
 func coreCounter(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		countCore(reqid.CoreIDFromContext(req.Context()))
+		if coreID := req.Header.Get("Chain-Core-ID"); coreID != "" {
```

Comment:
I think we need it somewhere, otherwise the count of Cores seen will be off by 1.